app/src/lib/stores/app-store.ts

```diff
@@ -412,9 +416,9 @@ export class AppStore extends TypedBaseStore<IAppState> {
 
   private wireupIpcEventHandlers(window: Electron.BrowserWindow) {
     ipcRenderer.on(
-      'window-state-changed',
-      (event: Electron.IpcMessageEvent, args: any[]) => {
-        this.windowState = getWindowState(window)
```

Comment:
While investigating this bug I also noted that the IPC call contains the `WindowState` we need to work with, so don't need to call this function if we can trust the value from the main process.